### PR TITLE
Isengard shortcuts

### PR DESCRIPTION
**Pull request by ghost**
Provides a version equivalency option to overrides.xml to check whether shortcuts/groupings with a <version/> tag apply to more than one version of Kodi. Makes it easier to manage shortcuts which change between Kodi versions. (Supersedes #15). Necessary because...

Switch available music library shortcuts to use Isengard Music Library Nodes. Also updates default shortcuts to use music library nodes on Isengard.

Two bugs that I came across whilst testing fixed also - invalid overrides.xml could lead to a build-loop, and defaultID set incorrectly on a new main menu when changing shortcut (may impact #17, but probably not).

### CODE REVIEW

**Comment by ghost** [comment]
I'm not convinced the version equivalency is the best way to handle shortcuts changing between Kodi versions, but it's the best I can come up with. If anyone has any alternative ideas... :)

**Comment by ghost** [comment]
I apologise for bumping a PR, but - as it appears no-one has any comments (and I presume, therefore, issues) regarding the version equivalency - is there anything I can do/change/whatever regarding this PR before it's pulled? I'd quite like to have the audio nodes in core for the code I'm working on currently :)

**Comment by BigNoid** [comment]
I can give you commit rights to speed things up :)

**Comment by ghost** [comment]
That's entirely up to you - though I'll be perfectly honest, it's been good to have my commits reviewed by a third party (Phil65 pointed out something that led to better code in #21, for example), rather than just making changes as I feel ;)

**Comment by ghost** [comment]
@BigNoid I see you have given me commit rights - and many thanks. I'll do my best to ensure I don't abuse it (as I say, having code reviewed by a third party is a genuinely useful thing) - but I'll enjoy using it to merge this initially, and probably a few updates to templates (a feature I'm using extensively, but finding a little limited), which I was going to save until Kodi J\* - but I'll at least give people a change to notice my mistakes before I commit ;)

File: resources/lib/datafunctions.py [hunk 1]
```diff
@@ -652,9 +671,59 @@ def checkVisibility ( self, action ):
             if path[ 1 ].endswith( ")" ):
                 path[ 1 ] = path[ 1 ][:-1]
             return NODE.get_visibility( path[ 1 ] )
-            
-        return ""
+
+        # Audio node visibility
+        elif action.startswith( "activatewindow(musiclibrary,musicdb://" ) or action.startswith( "activatewindow(10502,musicdb://" ) or action.startswith( "activatewindow(MusicLibrary,library://music/" ) or action.startswith( "activatewindow(10502,library://music/" ):
+            path = action.split( "," )
```
Comment (phil65) [inline resources/lib/datafunctions.py]: Is it safe to assume that every action contains a comma? If not then the next line would throw an exception.

Comment (ghost) [inline resources/lib/datafunctions.py:677]: [ignore me]

**Comment by phil65** [comment]
sorry, missed this one. :)
Feel free to ping me if I dont answer within a few days.
Can only do some basic code checking though, not familiar with all that code. This one looks fine from what I can judge.

**Comment by ghost** [comment]
The problem is I _am_ familiar with this code, and I really should have picked up on that. I'll ping you on the updated PR :)

**Comment by ghost** [comment]
Looking at the code, it's absolutely safe to assume they include a comma - the comma is there in line 676 - `activatewindow(musiclibrary,musicdb://` (all of the possibilities that line is checking for include a comma...)

**Comment by phil65** [comment]
yep, sorry, already noticed, comment was removed one minute later^^

**Comment by ghost** [comment]
No worries, and nice to know I actually got it right :>

**Pull request merged**